_posts/2021-11-03-flink-backward.md

```diff
@@ -0,0 +1,79 @@
+
+{% toc %}
+
+# A retrospective on the release cycle
+
+From the team, we collected emotions that have been attributed to points in time of the 1.14 release cycle:
+
+<center>
+<img src="{{site.baseurl}}/img/blog/2021-11-03-flink-backward/1.14-weather.png" width="70%"/>
+</center>
+
+The overall sentiment seems to be quite good. A ship crushed a robot two times, someone felt sick towards the end, an octopus causing negative emotions appeared in June...
+
+We looked at the origin of these emotions and came up with an analysis of what went well and what could be improved. We also incorporated some feedback gathered from the community.
+
+## Problems faced
+
+From a content perspective, the community is still ironing out processes around documentation and blog posts. There have been a lot of test instabilities and issues with some of the efforts to fix them. We also had to push the feature freeze by two weeks, which might actually be considered early and not actually be living up to the Apache Flink release tradition. :p
+
+## Things enjoyed
+
+The implementation of some features, such as [buffer debloating](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/memory/network_mem_tuning/#the-buffer-debloating-mechanism) and [fine-grained resource management](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/), went really smoothly. Though a few issues are now popping up. We enjoyed the moments while they last. :smiling_face_with_tear: 
+
+We also said goodbye to some components. It always feels refreshing to remove code and reduce complexity. :slightly_smiling_face:
+
+
+# What we want to achieve through process changes
+
+## Transparency - let the community participate
+
+When approaching a release (usually a couple of weeks after the previous release has been done) we set up bi-weekly meetings for the community to discuss any issues regarding the release. The usefulness of those meetings varied a lot and so we started to [track the efforts](https://cwiki.apache.org/confluence/display/FLINK/1.14+Release) in the Apache Flink Confluence wiki. 
+
+We came up with a system to label the current states of each feature: "independent", "won’t make it", "very unlikely", "will make it", "done", and "done done". We introduced the "done done" state since we were lacking a shared understanding of the definition of done. To qualify for "done done", the feature is manually tested by someone who has not been involved in the implementation of the specific effort and there exists comprehensive documentation that enables users to use the feature.
```

Comment:
```suggestion
We came up with a system to label the current states of each feature: “independent”, “won’t make it”, “very unlikely”, “will make it”, “done”, and “done done”. We introduced the “done done” state since we lacked a shared understanding of the definition of done. To qualify for “done done”, the feature is manually tested by someone not involved in the implementation. Additionally, there must exist comprehensive documentation that enables users to use the feature.
```